### ISSUE TITLE
Is it possible to use Motion Cam for RAW video?

### ISSUE THREAD

**Issue by akirayui** [mirsadm/motioncam#23]
Hello, 
Thank you for the project.

I was developing a similar app for personal use.
What I want to do is recording a 10-15 second DNG sequence in RAW10 format at 24FPS with 1/48 shutter speed. For this, I wanted to create a buffer and empty a certain part of it during shooting and reuse it. But I couldn't do it because it was too complicated for me.
Looks like you've already done the important part. Do you have any plans to develop your app so that it can be used for video as I mentioned?

**Comment by mirsadm** [comment]
What would you want to do with the 10-15 seconds of RAW10 data? Keep in mind on a 12 MP sensor 15 seconds worth of RAW10 data is over 5GB.

**Comment by locuturus** [comment]
SAF storage access framework could be used to set the save path to an SDcard or even USB storage. Open Camera has this as an option but defaults to saving under DCIM for simplicity. Assuming the storage device is fast enough to not choke this could improve the usefulness of taking even short clips since one might take a few of them at a go and rapidly fill internal storage.

**Comment by mirsadm** [comment]
I've managed to get it to work. On my device (OnePlus7 Pro) I can record about 30 seconds before running out of memory. The storage is not quite fast enough to be able to keep up in real time. Still 30 seconds of footage is over 10 GB. I need to write a small tool to convert the output into DNG then I'll upload a test version.

**Comment by locuturus** [comment]
Pixel 4a on Android 12. Turned off dual exposure, recorded ~9 sec (very laggy preview but no crash) and I see a zip with 177 frames inside. So far so good! No idea yet how it looks I'll have to get that all converted first.

**Comment by paolodepetrillo** [comment]
I got the raw2dng utility to build and run on Arch Linux using this [CMakeLists.txt](https://gist.github.com/paolodepetrillo/a3707e83f0dcdba4e34f6804112aaf78) file. The file goes in motioncam/raw2dng. I don't really know much about build systems and got this working by trial and error, so I'm sure there are lots of things done incorrectly.

`./setupenv.sh` must have been run successfully, and OpenCV must be installed, before building this.

On [this line](https://github.com/mirsadm/motioncam/blob/adcbdfe52f0ed825d308a02189befcb2ad755196/libMotionCam/libMotionCam/include/motioncam/CameraProfile.h#L30) in CameraProfile.h I had to replace `__unused` with `__attribute__((__unused__))` to get it to compile - apparently it's defined in a OSX header but not on Linux.

I am also testing with a Pixel 4a (not the newer 5G version). Only recorded a couple seconds to try it so far.

**Comment by akirayui** [comment]
The application you have published on Play Store runs without any problems, but this one crashes because Documents folder is not available by default on some Android devices like Xiaomi Redmi Note. If the user creates the directory manually,  "IOException" will thrown due to your application does not have permission to write to this directory. So I recommend using the DCIM folder reserved for media.

**Comment by mirsadm** [comment]
Yep, writing to the Document folder was always a hack just to test this feature. What I intend on doing is to save to the apps private storage area first then save the DNGs out to the DCIM folder afterwards.

**Comment by nongrata23** [comment]
hi, i also tried, i think is crushing because lack of permissions maybe?>

**Comment by owlgreens** [comment]
> I've put up a test build here: https://github.com/mirsadm/motioncam/releases
> 
> Couple of things to note:
> 
> 1. The app records a ZIP file to the internal storage Documents folder.
> 2. The ZIP file needs to be converted to a set of DNG files. I've written a tool that will do that which is here:
>    https://github.com/mirsadm/motioncam/tree/main/bin/mac
> 
> I've only compiled it for an x86 Mac for now. I'll look into creating a Linux binary as well. Otherwise the source code for it is here: https://github.com/mirsadm/motioncam/tree/main/raw2dng

Can you please repost the zip to dng conversion tool? I'm getting a 404. Thank you and keep up the great work!

**Comment by mirsadm** [comment]
I'm almost done with an update that will convert the files to DNG on the device itself after they are captured. The utility needs to be compiled at the moment. I will let you guys know when it's done.

**Comment by mirsadm** [comment]
@owlgreens I've uploaded a new release here
https://github.com/mirsadm/motioncam/releases/download/6.0.3-alpha0/MotionCam.6.0.3-alpha0.apk

It will convert the video to DNG on the device after capture

@akirayui I'm hoping this will also fix your problem with the Documents folder

**Comment by nongrata23** [comment]
Is possible to capture the frames into folders instead of zips?, also is possible to convert outside of the phone? because it takes a lot of time to convert even on a fast soc... but seems to be working!!

**Comment by eszdman** [comment]
>That's cropping from 4000x3000 (12MP) to 4000x2100 (8.4MP).
Add scaling also, to get 3840x2160
And software binning options for 2000x1050 as example
![2021-05-22 12_07_04-IMX378_Software_Reference_Manual_0 0 1 寄存 (3) pdf - Adobe Acrobat Pro](https://user-images.githubusercontent.com/45270494/142851635-6dde3ea0-22ce-43d5-8d9f-2f656e4e47f4.png)
![image](https://user-images.githubusercontent.com/45270494/142851731-c7f627db-cc9f-4746-96fb-25590e931269.png)

**Comment by akirayui** [comment]
The app doesn't crash anymore but I couldn't find where the video was saved.

**Comment by SkyDiverCool** [comment]
![image](https://user-images.githubusercontent.com/11019122/142911565-21e8d00a-fb0b-4442-a9c4-759fb29e62b1.png)
Found that the last handful of frames in a zip file are at no crop

**Comment by mirsadm** [comment]
New release here: [<...>](https://github.com/mirsadm/motioncam/releases/download/6.0.4-alpha0/MotionCam.6.0.4-alpha0.apk)

- That fixes the cropping issue for the last few frames
- I've added some more frame rate options (30, 24, 12, 5, 1)
- Focus is locked until clicking again on the focus circle
- Files are saved as DNGs in one directory, not in ZIP files anymore.

@eszdman The reason I've only added the cropping option is because it is very fast to do in real time. Scaling and binning would increase the chances of dropping frames. It is something you could do in post pretty easily though. Binning is an interesting idea but can also be achieved in post. If the purpose of binning is to reduce noise I can add an option to combine multiple frames.

**Comment by eszdman** [comment]
@mirsadm Scaling is fast, it can decrease number of subpixels but without cropping, so you should use it to save field of view

**Comment by mirsadm** [comment]
Relative to other image operations it is fast but every millisecond matters in this instance. Also skipping pixels (nearest neighbour scaling) will not result in a nice looking image. I'm wondering why you would like that though?

**Comment by nongrata23** [comment]
i have no words to express what i feel hhhhhhhhhh the world will burn in the following months, get ready hhhhhhh

**Comment by owlgreens** [comment]
> New release here: https://github.com/mirsadm/motioncam/releases/download/6.0.4-alpha0/MotionCam.6.0.4-alpha0.apk
> 
Gamechanging! Thank you for your incredible work!

**Comment by SpinozaWeb** [comment]
I am using LG v40 on android 10. I made a RAW video. but I don't know where the files are written. I don't have a documents folder on my phone. I even created a documents folder but there are no files there. Where does the program store files?

**Comment by eszdman** [comment]
> I'm wondering why you would like that though?

Because there are devices where the memory speed does not allow you to save 16 by 9, even taking into account the cropping of the edges due to the too high resolution of the source, scaling could help for this.

**Comment by nozorrog** [comment]
The raw video is stored in /data/data/ root folder. It's very inusefull and need to be resolved.

**Comment by Mrdoumhann** [comment]
> The raw video is stored in /data/data/ root folder. It's very unusefull and need to be resolved.

I think this folder is not accessible from non-rooted devices. Also there is no 
any other folder related to motioncam. So I could not find recorded DNGs.

**Comment by nozorrog** [comment]
> > The raw video is stored in /data/data/ root folder. It's very unusefull and need to be resolved.
> 
> I think this folder is not accessible from non-rooted devices. Also there is no any other folder related to motioncam. So I could not find recorded DNGs.

You are right. And I insiste to store writed files in DCIM folder to be accessed by user without problems.

**Comment by locuturus** [comment]
I suggest picking a user defined folder with SAF. Could be DCIM, Documents,
an SDcard, USB drive, whatever. My 2 cents,

**Comment by 87alex87** [comment]
External USB SSD (like Samsung T5) support would be great! :)

**Comment by xtrul** [comment]
Again i have no words to express myself :) i still cannot recover of excitement! still many things can be improved but the base is here now... i left you several messages let me know when u want can till then i enjoy this as it is

Regards
Sebastian

[Here is a sample](https://youtu.be/Ww4r0KUUYE4)

**Comment by mirsadm** [comment]
Update here:
https://github.com/mirsadm/motioncam/releases/download/6.0.5-alpha0/MotionCam.6.0.5-alpha0.apk

Videos are saved to Downloads folder now instead.

**Comment by Mrdoumhann** [comment]
> Update here: https://github.com/mirsadm/motioncam/releases/download/6.0.5-alpha0/MotionCam.6.0.5-alpha0.apk
> 
> Videos are saved to Downloads folder now instead.

Thanks for the update. But I can't find Downloads folder. Whats wrong with DCIM?

**Comment by mirsadm** [comment]
Storing them in the DCIM folder will result in every video frame showing up in Google Photos/image gallery which IMO is not good. Have you tried using a file manager? I'd be shocked if your phone doesn't have a Download folder

**Comment by xtrul** [comment]
is working fine, on downloads folder for s9+, lg g7, s20 fe^^

**Comment by Mrdoumhann** [comment]
![IMG_20211126_022153](https://user-images.githubusercontent.com/89611885/143507255-166e5196-46b5-4080-a1a0-22b6bc090852.jpg)

Yes Download folder is there but I think the app is looking for the "Download**s**"folder.

**Comment by mirsadm** [comment]
The app doesn't look for a Downloads or Download folder. It asks the OS to save it to wherever the Download folder is.

New release: here https://github.com/mirsadm/motioncam/releases/download/6.0.6-alpha0/MotionCam.6.0.6-alpha0.apk

- RAW videos don't get processed after capture now. They can manually be processed at a later time.
- If dual exposure is enabled in settings, RAW video capture mode provides a preview of the RAW image with a clipping indicator.
-  And finally the app shows the resolution and provides more cropping options.

**Comment by xtrul** [comment]
Thank you for this solid update! works great all features!

**Comment by Mrdoumhann** [comment]
> The app doesn't look for a Downloads or Download folder. It asks the OS to save it to wherever the Download folder is.

There are enough testers already, so just ignore this issue for now. Development is going very well, keep it up.

**Comment by mirsadm** [comment]
One final update for a while here: https://github.com/mirsadm/motioncam/releases/download/6.0.7-alpha0/MotionCam.6.0.7-alpha0.apk

@Mrdoumhann I've changed it so it lets you select where to save the RAW videos rather than going to Documents/Download. Hopefully that fixes your issue.

**Comment by Mrdoumhann** [comment]
> One final update for a while here: https://github.com/mirsadm/motioncam/releases/download/6.0.7-alpha0/MotionCam.6.0.7-alpha0.apk
> 
> @Mrdoumhann I've changed it so it lets you select where to save the RAW videos rather than going to Documents/Download. Hopefully that fixes your issue.

 Yes, problem solved. Finally I am able to see the holy dngs. Thank you for your effort.

**Comment by sx3z56** [comment]
> One final update for a while here: https://github.com/mirsadm/motioncam/releases/download/6.0.7-alpha0/MotionCam.6.0.7-alpha0.apk
> 
> @Mrdoumhann I've changed it so it lets you select where to save the RAW videos rather than going to Documents/Download. Hopefully that fixes your issue.

Hey! thank you very much for your work! it gives new life to mobile video shooting!I have a question. at the moment, your crop is implemented only along the x-axis. do you plan to add a crop on the y-axis? the thing is, my phone's main sensor has an 27 megapixels. and even with the maximum memory allocation, the buffer is clogged in seconds. a crop on the other axis would help a lot.

**Comment by mirsadm** [comment]
Yep I've added both horizontal/vertical cropping and fixed an issue with purple DNGs. Just need a bit more time to sort it out.

**Comment by experimentator1709** [comment]
> Yep I've added both horizontal/vertical cropping and fixed an issue with purple DNGs. Just need a bit more time to sort it out.

Hey! Please add 48, 50 and 60 frames per second. Another question is whether it is possible to record sound, at least in WAV. Thanks!

**Comment by xtrul** [comment]
> > Yep I've added both horizontal/vertical cropping and fixed an issue with purple DNGs. Just need a bit more time to sort it out.
> 
> Hey! Please add 48, 50 and 60 frames per second. Another question is whether it is possible to record sound, at least in WAV? Thanks!

48 50 60 wont work! unless the oem set it free

**Comment by experimentator1709** [comment]
> > > Yep I've added both horizontal/vertical cropping and fixed an issue with purple DNGs. Just need a bit more time to sort it out.
> > 
> > 
> > Hey! Please add 48, 50 and 60 frames per second. Another question is whether it is possible to record sound, at least in WAV? Thanks!
> 
> 48 50 60 wont work! unless the oem set it free

Thanks for the information. But suddenly the author succeeds :)

**Comment by xtrul** [comment]
Thank you! @mirsadm(for your brains) @akirayui (for yourvision) let the revolution begin!

https://youtu.be/slUvVN3VqJY

**Comment by mirsadm** [comment]
Update here https://github.com/mirsadm/motioncam/releases/download/6.0.8-alpha0/MotionCam.6.0.8-alpha0.apk

- Horizontal/vertical cropped
- Purple DNGs fix
- UI improvements 
- AE/AWB lock and OIS toggle

**Comment by mirsadm** [comment]
Update here https://github.com/mirsadm/motioncam/releases/download/7.0.1/MotionCam.7.0.1.apk
AF locking and some bug fixes

**Comment by xtrul** [comment]
> > Update here https://github.com/mirsadm/motioncam/releases/download/7.0.1/MotionCam.7.0.1.apk AF locking and some bug fixes
> 
> the buttons are so small

Lets finish functionality first :) UI after

**Comment by yousaf465** [comment]
you can edit the resulting dng directly in Da Vinci resolve by unchecking "Show Individual Frames" https://beginnersapproach.com/davinci-resolve-edit-images-stills/

**Comment by mirsadm** [comment]
https://github.com/mirsadm/motioncam/releases/download/7.0.2/MotionCam.7.0.2.apk

Just a small update to make the buttons a little bit easier to click

**Comment by experimentator1709** [comment]
> https://github.com/mirsadm/motioncam/releases/download/7.0.2/MotionCam.7.0.2.apk
> 
> Just a small update to make the buttons a little bit easier to click

HI! Which API did all cameras become available?

**Comment by mirsadm** [comment]
They're not really available, the app guesses and tries to open more cameras than reported by the system. This usually let's it use more of the cameras than reported by the system

**Comment by experimentator1709** [comment]
What is the quality of the JPG preview inside the DNG? Can we save the size of the DNG file? For example 50% JPG quality for preview?

**Comment by Automatisator100** [comment]
Hello Mirsad,

this is a super project (many are waiting for it!) and it is discussed in cinematography forums. 

The Huawei smartphones have a RYYB sensor, could you implement a conversion to RGGB before writing to DNG? That would be great. :) Because then the DNG from huawei can be edited directly in Davinci.

Best regards

**Comment by julienacquaviva** [comment]
Hello Mirsad,

Thanks a lot for you work, RAW video works very very well.
Although, I have hot pixels on my DNGs. Is there something fixable? Thanks in advance

![Untitled-1](https://user-images.githubusercontent.com/5604357/144746770-a12cab5a-df70-40a9-a6e6-5b9affc3c4d0.jpg)

**Comment by mirsadm** [comment]
@experimentator1709 do you mean reducing the size of the DNG? I can enable compression when creating the DNGs. It will make the process take longer though. Eventually I want to be able to do the conversion on a PC to reduce the time it takes.

@Automatisator100 It is certainly possible though I don't have access to a Huawei phone so it would be difficult for me to test.

@julienacquaviva I think those dots might from the auto focus system. Try the release below and set the frame merging to 4 or higher. If that doesn't fix it then I'll add the hot pixel removal that I use for photos.

Update here:
https://github.com/mirsadm/motioncam/releases/download/7.0.3/MotionCam.7.0.3.apk

- Option to merge frames to reduce noise when converting RAW video
- UI fixes

**Comment by experimentator1709** [comment]
@mirsadm Thank you! I'm waiting for the conversion to PC

**Comment by experimentator1709** [comment]
@mirsadm Can remove vignette be added to RAW metadata?

**Comment by julienacquaviva** [comment]
@mirsadm thanks for that quick answer, frame merging doesn't fix it though.
First I thought it was focus pixels then I tried Rawtherapee and applied the "hot pixels" fix and it worked.
Btw those dots come with the RAW photo mode as well (I mean Motion Cam's photo mode) but not with RAW photo from the stock camera.

**Comment by Automatisator100** [comment]
> @Automatisator100 It is certainly possible though I don't have access to a Huawei phone so it would be difficult for me to test.


I would offer myself as a tester. :)

But I think it is a bug. When I activate "Enable dual exposure controls", I have a picture like this:

https://drive.google.com/file/d/1S4wwXY4UbseGyk9TWY7mRO3oa9reH1cn/view?usp=sharing

if I deactivate "dual exposure controls" everything fits:

https://drive.google.com/file/d/1S0WdjdZDr1_F2_mZbombb3roFSzm4ZvD/view?usp=sharing

And when I use RAW VIDEO, it looks exactly the same as in PHOTO mode with "Enable dual exposure controls" enabled.

**Comment by mirsadm** [comment]
@Automatisator100 It is a bug in the sense that it doesn't support that bayer layout. It only supports RGGB/BGGR/GRGB/GBGR. When you enable the dual exposure controls it doesn't use the Android native preview, it generates one from the RAW images.

Update here with general UI fixes:
https://github.com/mirsadm/motioncam/releases/download/7.0.4/MotionCam.7.0.4.apk

**Comment by Automatisator100** [comment]
> @Automatisator100 It is a bug in the sense that it doesn't support that bayer layout. It only supports RGGB/BGGR/GRGB/GBGR. When you enable the dual exposure controls it doesn't use the Android native preview, it generates one from the RAW images.

Huawei P30(Pro) is equipped with three sensors and only one of them, the 40MP sensor, is RYYB.
The sensor used by motioncam is the middle 16MP sensor and as far as I know, that is not a RYYB sensor.

**Comment by Automatisator100** [comment]
> @Automatisator100 It is a bug in the sense that it doesn't support that bayer layout. It only supports RGGB/BGGR/GRGB/GBGR. When you enable the dual exposure controls it doesn't use the Android native preview, it generates one from the RAW images.

Can you maybe add a function in motioncam to select other cameras on Huawei p30 (Pro)? Because as it looks, only one sensor of the three is a RYYB sensor, but the other two are not. 

See also: https://developer.huawei.com/consumer/en/doc/development/Media-Guides/3180101

**Comment by experimentator1709** [comment]
@mirsadm 

Hey! DNG converter on Windows operating system. When approximately? :))

**Comment by julienacquaviva** [comment]
That would be awesome!

**Comment by mirsadm** [comment]
Yep I intend on doing this, probably not windows specific but in a browser with webassembly.

**Comment by julienacquaviva** [comment]
Thanks a lot. Is the 900 frames limit fixable for continuous recording?

**Comment by mirsadm** [comment]
Yeah I'll make it much higher.

**Comment by julienacquaviva** [comment]
Cool, thanks!

**Comment by julienacquaviva** [comment]
Also, the fps in the "manage videos" menu is incorrect. Or maybe it is and then the fps selection is not always effective.

**Comment by eszdman** [comment]
> I think those dots might from the auto focus system. Try the release below and set the frame merging to 4 or higher. If that doesn't fix it then I'll add the hot pixel removal that I use for photos.

You should use shielded pixel correction it's similar to lens shading map
![photo_2021-12-11_15-34-48](https://user-images.githubusercontent.com/45270494/145676691-b449ee15-7105-47e9-939c-3ab884d5a60b.jpg)

**Comment by experimentator1709** [comment]
@mirsadm 
https://github.com/experimentator1709/RawBayer2DNG

**Comment by mirsadm** [comment]
Update here:
https://github.com/mirsadm/motioncam/releases/download/7.0.8-beta1/MotionCam.7.0.8-beta1.apk

This lets you choose where to save the recorded videos.

@experimentator1709 I use a custom format that's compressed so that project won't work if you're intending on using it to convert the RAW images from the recorded ZIP file.

**Comment by 2blackbar** [comment]
This is amazing, can you make it output raw dng videos to fast SSD through usb3 cable with maximum speed  ?

**Comment by xtrul** [comment]
is allready

**Comment by yousaf465** [comment]
> Great but what converter do i need for raw to dng conversion ? I have raw2dng.exe for windows but it does not convert these raw files , rawbayer2dng crashes on phone i tried conversion to dng but it says queued and nothing moves. OK its a bug ! previous version 3 has it working, new one has it broken.It jus doesnt show location to save dngs after tapping convert to dng Please dont pack the files to zip ! Leave them in dng or raw in folder So does it go to ram memory when using ssd or goes straight to ssd over the cable ? What i mean is not using ram that much and offloading frames to ssd with usb3 speed.This would be very nice, cause its going to fill the phone very fast when storing to internal, sd card speeds arent that good but usb3 speed to ssd is enough for 4k raw video but is that possible to do in this app?

you can use lightroom, adobe camera raw, or any equivalent program.

**Comment by FilipZamec** [comment]
> Also, the fps in the "manage videos" menu is incorrect. Or maybe it is and then the fps selection is not always effective.

True for me too. I can see numbers like 26 fps, 12fps. And I always record at 24fps.

**Comment by FilipZamec** [comment]
> Great but what converter do i need for raw to dng conversion ? I have raw2dng.exe for windows but it does not convert these raw files , rawbayer2dng crashes on phone i tried conversion to dng but it says queued and nothing moves. OK its a bug ! previous version 3 has it working, new one has it broken.It jus doesnt show location to save dngs after tapping convert to dng Please dont pack the files to zip ! Leave them in dng or raw in folder So does it go to ram memory when using ssd or goes straight to ssd over the cable ? What i mean is not using ram that much and offloading frames to ssd with usb3 speed.This would be very nice, cause its going to fill the phone very fast when storing to internal, sd card speeds arent that good but usb3 speed to ssd is enough for 4k raw video but is that possible to do in this app?

I have the same problem. On latest versions, sometimes when I want to convert the video into dngs, I get message QUEUED and nothing moves and nothing is happening. Sometimes it works.

I have Mi 11 Ultra. Snapdragon 888.
Thanks for the great app. RAW video is jaw dropping :)

**Comment by FilipZamec** [comment]
> > Also, the fps in the "manage videos" menu is incorrect. Or maybe it is and then the fps selection is not always effective.
> 
> True for me too. I can see numbers like 26 fps, 12fps. And I always record at 24fps.
![wrongfps](https://user-images.githubusercontent.com/34546948/146470230-e5aeeb6e-6947-48a6-92b8-f89fcdff411d.jpg)

I recorded 10s video in 24fps and the output was really only 129 frames. It should be twice as much. Happens sometimes in newer versions. In 6.0.8 never happened to my knowledge.

**Comment by FilipZamec** [comment]
One last comment. Captured 3 videos in a row with the same settings.

This time, frame count is correct, only fps label is wrong.

![wrongfps2](https://user-images.githubusercontent.com/34546948/146470793-c3051fc1-6c21-4b3e-bf58-d805319ac735.png)

**Comment by yousaf465** [comment]
> Great but what converter do i need for raw to dng conversion ? I have raw2dng.exe for windows but it does not convert these raw files , rawbayer2dng crashes on phone i tried conversion to dng but it says queued and nothing moves. OK its a bug ! previous version 3 has it working, new one has it broken.It jus doesnt show location to save dngs after tapping convert to dng Please dont pack the files to zip ! Leave them in dng or raw in folder So does it go to ram memory when using ssd or goes straight to ssd over the cable ? What i mean is not using ram that much and offloading frames to ssd with usb3 speed.This would be very nice, cause its going to fill the phone very fast when storing to internal, sd card speeds arent that good but usb3 speed to ssd is enough for 4k raw video but is that possible to do in this app?

https://youtu.be/0ii0WuS6QR8 I have created this video, how to import the DNG video/sequence created by the motioncam app into Davinci resolve.

**Comment by mirsadm** [comment]
Thanks for the video above. I've made some signifcant improvements here:
https://github.com/mirsadm/motioncam/releases/download/8.0.0-beta0/MotionCam.8.0.0-beta0.apk

**Comment by julienacquaviva** [comment]
@mirsadm thanks but OIS switch bas disappeared, manage menu is crashing and preview framing is square not 4:3 anymore

**Comment by 87alex87** [comment]
> Thanks for the video above. I've made some signifcant improvements here: https://github.com/mirsadm/motioncam/releases/download/8.0.0-beta0/MotionCam.8.0.0-beta0.apk

Hi. It seems like vertical crop setting affects horizontal crop in preview and vice versa :)

**Comment by mirsadm** [comment]
haha yeah I noticed it as soon as I posted it. I'll fix it later today. @julienacquaviva my guess is that if you clear out any of your existing videos it will fix that but if you don't want to do that I'll will hopefully be able to fix it.

**Comment by julienacquaviva** [comment]
@87alex87 I see, thanks!
@mirsadm Alright, thanks for the hard work!

**Comment by mirsadm** [comment]
Crop marks fixed
https://github.com/mirsadm/motioncam/releases/download/8.0.0-beta1/MotionCam.8.0.0-beta1.apk

**Comment by julienacquaviva** [comment]
@mirsadm Thanks! I was wondering if a fix for stable constant FPS would be doable? I always get frame drops and with the V8 the FPS output displayed when recording is never stable. Thanks

**Comment by 2blackbar** [comment]
yes, get ssd, your phones internal sd is not weiting fast enough, check it with sd bench , should be ober 300mb speed to deal with 4k full res

**Comment by julienacquaviva** [comment]
@2blackbar thanks but my internal memory is faster enough, around 450Mo/sec. It's something with the app, it's not constant framerate. It's like 24.2 to 26.8 fps.

**Comment by mirsadm** [comment]
The UI is not clear. That is the writing speed not the final frame rate. It's always going vary as the writing speeds are not consistent. What I shall add is the actual recorded FPS as well

**Comment by FilipZamec** [comment]
I got it too. Plenty of write speed over 400MB/s and fps fluctuates. Sometimes I set 24 fps and got 30 fps. It's showing 30 during recording and really the result is 30 not 24.
So it's not about slow storage 🙂

**Comment by julienacquaviva** [comment]
I know it's pretty difficult to change from variable to constant framerate on Android when recording video. Mcpro24fps application has a fix for constant flow, I don't know if it could be of any help because it works well.

**Comment by 2blackbar** [comment]
I think ideally a  setting with ratio between amount of frames stored in ram vs amount of frames saved to card could be useful, so we would set more aggressive settings for saving on internal sd more frames and more often so ram wont get full so fast.
Maybe saving 20 frames at once  to internal sd is better than saving one frame at a time.
Card spanning ( saving one frame internally , one frame to external sd) would also help to offload ram.
But overall i think more pressure should be put onto sd to try to offload ram as fast as possible.
